Makefile

```diff
@@ -1,22 +1,23 @@
-
+DESTDIR ?= 
+PREFIX ?= /usr/local/ 
 BUILDDIRS=src lib tool doc test
 
 .PHONEY: all install clean distclean
 
 all:
-	@$(foreach dir,$(BUILDDIRS),make -C '$(dir)' all && ) echo Built all
+	@$(foreach dir,$(BUILDDIRS),make DESTDIR=$(DESTDIR) PREFIX=$(PREFIX) -C '$(dir)' all && ) echo Built all
 	@echo '#'
 	@echo '# NOTE: You may need to "make testdata" for some tool/* programs'
 	@echo '#'
 
 install: all
-	@$(foreach dir,$(BUILDDIRS),make -C '$(dir)' install && ) echo Installed
+	@$(foreach dir,$(BUILDDIRS),make -e DESTDIR=$(DESTDIR) PREFIX=$(PREFIX) -C '$(dir)' install && ) echo Installed
 
 uninstall:
-	@$(foreach dir,$(BUILDDIRS),make -C '$(dir)' uninstall && ) echo Uninstalled
+	@$(foreach dir,$(BUILDDIRS),make -e -C '$(dir)' uninstall && ) echo Uninstalled
```

Comment:
Why do you need to override *all* envvars with `-e`?  That could lead to problems such as build instabilities; it is usually better to rely on `+=` and `if` etcetera.